### ISSUE TITLE
jsx-indent-props: Enforcing incorrect indents when using a component with props in ternary before

### ISSUE THREAD

**Issue by tobiaswaltl** [yannickcr/eslint-plugin-react#3146]
I experience some odd behavior of `jsx-indent-props` depending on a ternary before the actual component.
When having a component with props within the ternary and not having the whole ternary in one line, the rule enforces more indents for the subsequent component than it should.

I would expect the ternary not having any impact on the returning component.

Tested versions: 7.27.0 and 7.27.1

```
import React from 'react';

// works as expected - no props for the component in ternary
export const LintExample = () => {
  const foo = true
    ? <div>test</div>
    : false;

  return <div
    id="id"
  >
    test
  </div>;
};

// works as expected - props for the component in ternary but ternary in one line
export const LintExample1 = () => {
  const foo = true ? <div id="id">test</div> /* added prop "id" here*/ : false;

  return <div
    id="id"
  >
    test
  </div>;
};

// does not work as expected - props for the component in ternary and ternary not in one line
export const LintExample2 = () => {
  const foo = true
    ? <div id="id">test</div> // added prop "id" here
    : false;

  return <div
    id="id" // linter complains: ESLint: Expected indentation of 6 space characters but found 4.(react/jsx-indent-props)
  >
    test
  </div>;
};
```

**Comment by ljharb** [comment]
Can you provide your rule config?

**Comment by tobiaswaltl** [comment]
My rule config is as follows (retrieved via `eslint --print-config lint-example.tsx` so there shouldn't be any overrides in the cascade that I might have missed) :

```
"react/jsx-indent-props": [
      "error",
      2
    ],
```

**Comment by ljharb** [comment]
Interesting; I'm having trouble reproducing your issue - in other words, all three examples are failing the rule in my local tests.

**Comment by tobiaswaltl** [comment]
Weird, I just double-checked the example and in my case only the 3rd one fails the rule.
Version of `eslint-plugin-react` is `7.27.1`.
Version of `eslint` is `7.32.0`. I also just tried it with the current version `8.3.0` and again only the 3rd example failed.

Do you agree that the expected behavior would be that none of the examples fails the rule for the given rule config?

**Comment by ljharb** [comment]
I'm not sure what the expected behavior should be; I would expect multiline jsx to always be wrapped in parens, so it'd look like this:
```jsx
export const LintExample2 = () => {
  const foo = true
    ? <div id="id">test</div> // added prop "id" here
    : false;

  return (
    <div
      id="id"
    >
      test
    </div>
  );
};
```

**Comment by tobiaswaltl** [comment]
With parens, everything is fine. However, the `LintExample2` without parens is also valid syntax. So in that case I would expect that the props have to be indented by one level while the rule asks me to indent it by two levels.

**Comment by ljharb** [comment]
I see what you're saying. While I find that indentation style horribly confusing, it does seem like it should be accepted. A PR with failing test cases, and ideally a fix, would be appreciated.

**Comment by tobiaswaltl** [comment]
Ok, great! I'll try to submit one by the end of next week.

**Comment by Kegulf** [comment]
This now breaks several indent uses without any JSX in JS-files in my project
<img width="450" src="https://user-images.githubusercontent.com/24886567/155683794-16ea5f9a-89b6-4c56-850d-51b75e4cf253.png" />
<img width="550" src="https://user-images.githubusercontent.com/24886567/155683815-c01fc8ee-b76f-4dbe-9017-22432f6a8008.png" />

<img width="350" src="https://user-images.githubusercontent.com/24886567/155683960-ddb9f8d0-69b8-48bc-a41e-55c197d99928.png" />
<img width="550" src="https://user-images.githubusercontent.com/24886567/155684008-170fee0d-2e31-4b9e-824a-d83b2f84306e.png" />

Also breaks this type of formated ternaries with JSX:
<img width="400" src="https://user-images.githubusercontent.com/24886567/155684369-d2650e06-c7ce-4066-adc6-0c9218c3fb8c.png" />
<img width="550" src="https://user-images.githubusercontent.com/24886567/155684469-501ae050-678a-439a-a36a-375852f98c65.png" />

**Comment by Kegulf** [comment]
@ljharb @tobiaswaltl

**Comment by Kegulf** [comment]
Still broken in 7.29.0

Sorry if that first message felt harsh btw 😅

**Comment by ljharb** [comment]
@Kegulf you say "still", but this change was first released in v7.2.9.0, about 12 hours ago. A fix is coming shortly; see #3218.
